src/payment/strategies/paypal-commerce/paypal-commerce-script-loader.ts

```diff
@@ -15,8 +15,8 @@ export default class PaypalCommerceScriptLoader {
         this._window = window;
     }
 
-    async loadPaypalCommerce(options: PaypalCommerceScriptOptions): Promise<PaypalCommerceSDK> {
-        if (!options || !options.clientId || !options.merchantId) {
+    async loadPaypalCommerce(options: PaypalCommerceScriptOptions, isProgressiveOnboardingAvailable?: boolean): Promise<PaypalCommerceSDK> {
+        if (!options || !options.clientId || (!options.merchantId && !isProgressiveOnboardingAvailable)) {
             throw new InvalidArgumentError();
```

Comment:
Might be worth throwing an exception with a description about the error, e.g:
```js
throw new InvalidArgumentError('Unable to proceed because "merchantId" is required.');

Reply:
@capsula4 Good idea. Thank you!